### PR TITLE
Generate NRE docs for website

### PR DESCRIPTION
**Pull request by flaviut**
- add nre to `website.ini`
- add nre to `lib.txt`
- modify `tools/nimweb.nim` to handle cases here a module is both a directory and a `.nim` file.
  - that seemed to have broken the source links, so I also unbroke them.
- Update the NRE documentation, fixing RST mistakes and adding an example.

Thanks to Konstantin Molchanov for getting me to look into this!

### CODE REVIEW

File: lib/impure/nre.nim [hunk 1]
```diff
@@ -24,46 +24,35 @@ from unicode import runeLenAt
...
-## > 3. Altered versions must be plainly marked as such, and must not be
-## >    misrepresented as being the original software.
-## >
-## > 4. If PCRE is embedded in any software that is released under the GNU
-## >    General Purpose Licence (GPL), or Lesser General Purpose Licence (LGPL),
-## >    then the terms of that licence shall supersede any condition above with
-## >    which it is incompatible.
+## .. _`some additional terms`: http://pcre.sourceforge.net/license.txt
+##
+## Example
```
Comment (dom96) [inline lib/impure/nre.nim]: Could we get more specific examples instead of just a large single example?

File: tools/nimweb.nim [hunk 2]
```diff
@@ -165,10 +165,10 @@ proc walkDirRecursively(s: var seq[string], root, ext: string) =
 
 proc addFiles(s: var seq[string], dir, ext: string, patterns: seq[string]) =
   for p in items(patterns):
+    if existsFile(dir / addFileExt(p, ext)):
```
Comment (dom96) [inline tools/nimweb.nim]: what are the changes in this file about?

Comment (flaviut) [inline tools/nimweb.nim]:
```
     if existsDir(dir / p): # this branch is taken if a directory named dir/p exists
      walkDirRecursively(s, dir / p, ext)
    else:  # so when both a file dir/p.nim and dir dir.p exist, this branch never gets executed, even though it should
      add(s, dir / addFileExt(p, ext))
```

File: tools/nimweb.nim [hunk 3]
```diff
@@ -276,18 +276,18 @@ proc buildDoc(c: var TConfigData, destPath: string) =
     commands = newSeq[string](len(c.doc) + len(c.srcdoc) + len(c.srcdoc2))
     i = 0
   for d in items(c.doc):
-    commands[i] = "nim rst2html $# --docSeeSrcUrl:$#/$#/$# -o:$# --index:on $#" %
-      [c.nimArgs, c.gitRepo, c.gitCommit, d.pathPart,
+    commands[i] = "nim rst2html $# --docSeeSrcUrl:$#/$# -o:$# --index:on $#" %
+      [c.nimArgs, c.gitRepo, c.gitCommit,
```
Comment (flaviut) [inline tools/nimweb.nim]: The `d.pathPart` of the URL was getting doubled up, eg. `https://github.com/nim-lang/Nim/blob/devel/lib/impure/lib/impure/nre.nim`. Removing it here fixes that. I ran a tool to look for regressions (github 404s), and it was unable to find any.

File: lib/impure/nre.nim (hunk 1 above)
Comment (flaviut) [inline lib/impure/nre.nim]: Good idea, it's just going to take a little more time.

File: tools/nimweb.nim (hunk 3 above)
Comment (Araq) [inline tools/nimweb.nim]: I remember adding that, so maybe it's some subtle Win/Unix difference going on here. We need to investigate further.

Comment (dom96) [inline tools/nimweb.nim]: Does nimweb handle url's using ``os`` module's path procs? If so, it shouldn't.

Comment (Araq) [inline tools/nimweb.nim]: Who knows ... We need to investigate :P

**Pull request merged**